models/user.go

```diff
@@ -263,11 +263,11 @@ func (u *User) RelAvatarLink() string {
 	return setting.GravatarSource + u.Avatar
 }
 
-// AvatarLink returns user gravatar link.
+// AvatarLink returns user avatar link.
 func (u *User) AvatarLink() string {
 	link := u.RelAvatarLink()
 	if link[0] == '/' && link[1] != '/' {
-		return setting.AppSubUrl + link
+		return strings.TrimSuffix(setting.AppUrl, "/") + link
```

Comment:
I see, well the code still takes the path of least unpredictability.
It is locally known that the "link" variable starts with a slash,
and it is known that an URL would be valid either with and w/out
and ending slash, so it's just safer to avoid the risk.

In absence of a dedicated testsuite, I'd rather avoid risks.
Someone changing the code in setting.go could hardly guess
some code in user.go was dependent on AppUrl having an enforced
ending slash.

Go strings have known length, so the additional cost of TrimSuffix
over chopping the initial portion of relative avatar link is that
of comparing a character for each slash found in AppUrl.

Is that really worth the time we're spending on discussing this ?